yarn-project/acir-simulator/src/simulator.ts

```diff
@@ -44,12 +45,29 @@ export class AcirSimulator {
     return execution.run();
   }
 
+  // TODO Should be run as unconstrained function
   public computeNoteHash(notePreimage: Fr[], bbWasm: BarretenbergWasm) {
     return pedersenCompressInputs(bbWasm, [NOTE_PEDERSEN_CONSTANT.toBuffer(), ...notePreimage.map(x => x.toBuffer())]);
   }
 
+  // TODO Should be run as unconstrained function
   public computeNullifier(notePreimage: Fr[], privateKey: Buffer, bbWasm: BarretenbergWasm) {
     const noteHash = this.computeNoteHash(notePreimage, bbWasm);
     return pedersenCompressInputs(bbWasm, [NULLIFIER_PEDERSEN_CONSTANT.toBuffer(), noteHash, privateKey]);
   }
+
+  // TODO Should be run as unconstrained function
```

Comment:
TODO: Should be moved to circuits.js so that it can be side by side with the c++ code, where the logic is defined.

Reply:
True! The siloing makes more sense in circuits.js